gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineJavaVersions.java

```diff
@@ -39,6 +40,10 @@
     public static final String EXTENSION_NAME = "javaVersions";
 
     public static final GradleVersion MIN_GRADLE_VERSION = GradleVersion.version("7.0");
+    // 'nebula.maven-publish' and 'com.github.johnrengelman.shadow' create publications lazily which cause
+    // inconsistencies based on ordering.
+    private static final ImmutableSet<String> LIBRARY_PLUGINS =
+            ImmutableSet.of("nebula.maven-publish", "com.github.johnrengelman.shadow");
```

Comment:
Is it actually the `johnrengelman` shadow or is it our `gradle-shadow-jar` which does this?

Reply:
You're right, I think this is on us due to ShadowJarPlugin `ensureShadowJarIsOnlyArtifactOnJavaConfigurations` + `dependOnJarTaskInOrderToTriggerTasksAddingManifestAttributes`